paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/StartingScanner.java

```diff
@@ -37,14 +37,17 @@ public interface StartingScanner {
     class Result {
```

Comment:
Can we just return a interface, and scanner returns different implement classes.